pkg/api/core/v1/models/app.go

```diff
@@ -30,8 +30,31 @@ const (
 	ApplicationStagingActive = "active"
 	ApplicationStagingDone   = "done"
 	ApplicationStagingFailed = "failed"
+
+	ProviderGit              = "git"
+	ProviderGithub           = "github"
+	ProviderGithubEnterprise = "github_enterprise"
+	ProviderGitlab           = "gitlab"
+	ProviderGitlabEnterprise = "gitlab_enterprise"
```

Comment:
I was thinking that it will be better to have this as a `type` and change the provider to be a `GitProvider` instead of a generic string. This will help us on being stricter during the creation/handling of this value, and be sure to use one of the right one.

Something like:

```golang
type GitProvider string

const (
	ProviderUnknown = GitProvider("unknown")
	ProviderGit     = GitProvider("git")
	ProviderGithub  = GitProvider("github")
)

func GitProviderFromString(provider string) (GitProvider, error) {
	switch provider {
	case "git":
		return ProviderGit, nil
	case "github":
		return ProviderGit, nil
	default:
		return ProviderUnknown, errors.New("unknown provider")
	}
}

```

I really miss enums sometimes.

Reply:
Certainly doable. The conversion function acts as the validator.